packages/api/src/context.ts

```diff
@@ -1,39 +1,79 @@
-// src/server/router/context.ts
-import * as trpc from "@trpc/server";
-// import type { FetchCreateContextFnOptions } from "@trpc/server/adapters/fetch";
-import * as trpcNext from "@trpc/server/adapters/next";
-import { prisma } from "@agreeto/db";
+import type { inferAsyncReturnType } from "@trpc/server";
+import type { CreateNextContextOptions } from "@trpc/server/adapters/next";
+// import { getServerSession } from "@acme/auth";
+import { getToken, type JWTToken } from "@agreeto/auth";
+import { prisma, type User } from "@agreeto/db";
 
-/**
- * Replace this with an object if you want to pass things to createContextInner
- */
-type CreateContextOptions = Record<string, never>;
+export type CreateContextOptions = {
+  token: JWTToken | null;
+  // session: Awaited<ReturnType<typeof getServerSession>>;
+  user: User;
+};
 
 /** Use this helper for:
- *  - testing, where we dont have to Mock Next.js' req/res
- *  - trpc's `createSSGHelpers` where we don't have req/res
- */
+ * - testing, so we dont have to mock Next.js' req/res
+ * - trpc's `createSSGHelpers` where we don't have req/res
+ * @see https://beta.create.t3.gg/en/usage/trpc#-servertrpccontextts
+ **/
 export const createContextInner = async (opts: CreateContextOptions) => {
   return {
     prisma,
+    token: opts.token,
+    // session,
+    user: opts.user,
   };
 };
 
 /**
  * This is the actual context you'll use in your router
  * @link https://trpc.io/docs/context
  **/
-export const createContext = async (
-  opts: trpcNext.CreateNextContextOptions
-) => {
-  return await createContextInner({});
+export const createContext = async ({ req }: CreateNextContextOptions) => {
+  console.log("secret: ", process.env.NEXTAUTH_SECRET);
+  const token = await getToken({ req });
+  console.log("|||||||||||||||||");
+  console.dir({ token }, { depth: 2 });
+  console.log("|||||||||||||||||");
+
+  // FIXME: grab a real user
+  const user = {
+    id: "1",
+    email: "a.b@c.com",
+    name: "A B",
+    emailVerified: new Date(),
+    image: "https://placekitten.com/200/300",
+  };
```

Comment:
should extract user from token instead, this was just to get all types and linter together